### PR TITLE
Pin protobuf version to fix regression

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by thiagocrepaldi** [comment]
Fixes https://github.com/microsoft/onnxruntime/issues/11663

**Comment by yan12125** [comment]
Changing requirement files used for CI may not be enough. Per setup.py, `requirements-training.txt` (for onnxruntime-training) and `requirements.txt.in` (for other packages) should probably be updated as well.

By the way, this PR overlaps with an earlier one: https://github.com/microsoft/onnxruntime/pull/11682

**Comment by thiagocrepaldi** [comment]
> Not sure using protobuf 3.20 for Python is a good idea or not. C++ codes are not yet compatible with protobuf 3.20 (#11129)

They were unsaved on my editor, my bad

**Comment by thiagocrepaldi** [comment]
> Not sure using protobuf 3.20 for Python is a good idea or not. C++ codes are not yet compatible with protobuf 3.20 (#11129)

If it doesn't work, we can go with an older one, but in theory all 3.X should be compatible as they belong to the same major version

**Comment by yan12125** [comment]
> If it doesn't work, we can go with an older one, but in theory all 3.X should be compatible as they belong to the same major version

Sure! Thanks for updating them all.

**Comment by justinchuby** [comment]
Lint is being fixed at https://github.com/microsoft/onnxruntime/pull/11704

File: cgmanifests/cgmanifest.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
       {
         "component": {
           "Type": "maven",
-          "maven": { "GroupId": "com.google.protobuf", "ArtifactId": "protobuf-java", "Version": "3.20.1" },
```
Comment (snnn) [inline cgmanifests/cgmanifest.json:22]: Then you will re-introduce the problem that was fixed in https://github.com/microsoft/onnxruntime/pull/10420 See: https://github.com/advisories/GHSA-wrvw-hg22-4m67

**Pull request closed without merging**